File: app/src/main/java/za/org/grassroot2/presenter/activity/GrassrootPresenter.java
package za.org.grassroot2.presenter.activity;

import retrofit2.Response;
import za.org.grassroot2.model.exception.AuthenticationInvalidException;
import za.org.grassroot2.model.exception.ServerUnreachableException;

public interface GrassrootPresenter {
    void handleResponseError(Response response);
    void handleNetworkConnectionError(Throwable t);
    void handleNetworkUploadError(Throwable t);
    void handleAuthenticationError(AuthenticationInvalidException t);
    void handleServerUnreachableError(ServerUnreachableException t);

}

File: app/src/main/java/za/org/grassroot2/presenter/activity/GrassrootPresenter.kt
package za.org.grassroot2.presenter.activity

import retrofit2.Response
import za.org.grassroot2.model.exception.AuthenticationInvalidException
import za.org.grassroot2.model.exception.ServerUnreachableException

interface GrassrootPresenter {
    fun handleResponseError(response: Response<*>)
    fun handleNetworkConnectionError(t: Throwable)
    fun handleNetworkUploadError(t: Throwable)
    fun handleAuthenticationError(t: AuthenticationInvalidException)
    fun handleServerUnreachableError(t: ServerUnreachableException)

}
